object/datastore.go

```diff
@@ -339,3 +339,13 @@ func (d Datastore) Stat(ctx context.Context, file string) (types.BaseFileInfo, e
 	return res.File[0], nil
 
 }
+
+// Type returns the type of file system volume.
+func (d Datastore) Type(ctx context.Context) (types.HostFileSystemVolumeFileSystemType, error) {
+	var mds mo.Datastore
+
+	if err := d.Properties(ctx, d.Reference(), []string{"summary"}, &mds); err != nil {
```

Comment:
Should be able to use "summary.type" here instead of "summary".  Small payload in this case, but good habit to limit the property collector.